### DISCUSSION TITLE
In what format to store pictures?

### DISCUSSION THREAD

**Discussion by DragonsWho** [DragonsWho/CYOACatalog#19] [Q&A]
As far as I remember right now the site does not compress images at all (I tried different variant, up to even minimal compression is noticeable on the text) 
but at the same time all images weigh a hell of a lot and it's bad not only because they take up space on the server, but also pages load terribly long. 
Can anyone have any ideas about this? Or should we store all images in png?

**Comment by pixelRealmCo** [comment]
The best image format that you can currently use is generly avif they have the best lossless compression. 
The other alternative is webp. webp is more compatible then avif across browser and diffrent tools. In my job we only use webp for example becuse we have an image optimization thing in our framework that don't accept avif files, but if you don't have any of those requirments then I would go with avif if possible

**Comment by DragonsWho** [comment]
Is there a lossless mode in there? Damn, I must be an idiot, lol. I'll have to try this all over again, because right now the pages are taking two minutes to load for me.

**Reply by pixelRealmCo** [reply]
I mean if you only specify the image that is supposed to be posted it will post that exact image, when you are compressing it with a tool you can often lower the quality of the image and it will "lose some data" but if you don't compress it and just go with keeping the quality and just transform the image to an avif then you will still save a lot data

**Comment by DragonsWho** [comment]
Okay, I've been doing a little experimenting to figure it out. 
Afiv is a great format, but in lossles mode it makes the files even bigger than they were, lol.

So tentatively I suggest saving everything in webp losless format, which will save ~20%.

and to improve the user experience, pre-load images to the page in avif format with reduced pallete ~140 colors, 0.01 dithering (or 0.04) and ~25% quality.   (somewhere between 15% and 30% I haven't decided exactly yet)  (these color settings reduce the quality but increase the clarity of the text)

with such settings avif weighs extremely little, reducing file weight by 95%. The picture quality, of course, deteriorates, but the text remains readable, at least on my test samples. So the page will load in a second and the player can immediately start enjoying CYOA, and when the picture is loaded in the original quality, it will replace this pre-view. 
 
 # Extra

avif also has a function to split the image into blocks, so that only those segments that are visible to the user can be loaded. If we manage to customize it, we can raise the quality of avif file up to 50%, but still the visible part of the page will be loaded instantly. 

I'm not sure about Effort settings yet, which would compress the image better. At high settings one picture is compressed for about 10 minutes, given that I have a fairly powerful laptop, but so the picture can be compressed even stronger. However, not much, for example, instead of 100 kilobytes the first screen will weigh 80 or something like that, so perhaps it can be neglected. Or try and configure the server that it would compress and replace pictures with improved versions when it has free resources (so far usually works only 10% of the processor)

**Comment by DragonsWho** [comment]
I think that this should be done when the CMS is changed, it is pointless to mess around with Strapi settings if in a week everything will have to be redone
